documentation/driver_object.md

```diff
@@ -0,0 +1,152 @@
+# OP-TEE Generic Driver Object
```

Comment:
If you write any other new (pure) documentation files from scratch, please write them in the [rst-format](http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) instead. The reason for that is that we are about to collect all documentation at a single place and all files will be in rst then. You couldn't possible now that, so now blame here and I'll do the conversion on my own later. In any case, thanks a lot for taking the timing writing this. It's always appreciated when engineers take the time to write real documentation.

Reply:
When new document format is going to be integrated? If I can help by doing it already in the correct format, let me now?
BTW, what will be the file extension?

Reply:
In general, having more doc is great but in this case I think most of what is written here is just a duplicate of what is (or should be) in the the code. This will turn into a maintenance burden. I'd rather have more comments in the header files and in the implementation, if needed.
A few lines to mention the existence of the API and maybe its overall principles is fine however (your Introduction section more or less).